src/hypofuzz/hypofuzz.py

```diff
@@ -100,40 +93,111 @@ def from_hypothesis_test(
+
+        self.random = Random()
+        self.state: Optional[HypofuzzStateForActualGivenExecution] = None
+        self.provider = HypofuzzProvider(None)
+        self.stop_shrinking_at = math.inf
+        self._fixturedefs: list[pytest.FixtureDef] = []
+
+    def _new_state(
+        self, *, extra_kwargs: Optional[dict[str, Any]] = None
+    ) -> HypofuzzStateForActualGivenExecution:
+        _, _, stuff = process_arguments_to_given(
+            self.wrapped_test,
+            arguments=(),
+            kwargs=(self.extra_kwargs or {}) | (extra_kwargs or {}),
+            given_kwargs=self.wrapped_test.hypothesis._given_kwargs,  # type: ignore
+            params=get_signature(self.wrapped_test).parameters,  # type: ignore
+        )
+        return HypofuzzStateForActualGivenExecution(  # type: ignore
             stuff,
-            self._test_fn,
+            self.test_fn,
             settings(
                 database=self.database._db,
                 deadline=None,
                 suppress_health_check=list(HealthCheck),
                 verbosity=Verbosity.quiet,
             ),
             self.random,
-            wrapped_test,
+            self.wrapped_test,
         )
-        self.wrapped_test = wrapped_test
-        self.pytest_item = pytest_item
 
-        self.provider = HypofuzzProvider(None)
-        self.stop_shrinking_at = math.inf
+    def _enter_fixtures(self) -> None:
+        if self.pytest_item is None:
+            self.state = self._new_state()
+            return
+
+        assert self.pytest_item is not None
+        assert self.fixtureinfo is not None
+        # iter_parents is from function -> session, and listchain is from
+        # session -> function. We want the latter here, as expected by
+        # pytest.Session. (also, iter_parents only exists from pytest 8.1.1.)
+        for parent in self.pytest_item.listchain():
+            # TODO can parent ever not be an Item? .setup expects an Item, but
+            # listchain returns Node. pytest seems to use the pytest_runtest_setup
+            # hook to actually call SetupState.setup(item), so I'm not sure what
+            # guarantees are made by listchain in this context.
+            if not isinstance(parent, Item):
+                continue
+            self.pytest_item.session._setupstate.setup(parent)
+
+        request: FixtureRequest = self.pytest_item._request
+        extra_kwargs = {}
+        for (
+            name,
+            fixturedefs,
+        ) in self.fixtureinfo.name2fixturedefs.items():
+            # If there are multiple fixtures with this name, pick the last one,
+            # which is the closest one to us in scope.
+            #
+            # I'm not sure this is correct. More tightly scoped fixtures
+            # override less tightly scoped fixtures in pytest, but I don't know
+            # if this correctly handles interactions like dependent fixtures
+            # higher up the scope stack being overridden.
```

Comment:
Seems like a good case for some tests, which can check that pytest and hypofuzz see the same sequence of executed fixtures and of resulting values.

Reply:
I'm afraid of the correctness rabbit hole I'll find when I write tests for this 😅

Reply:
Wrote one test that seems ok, though I'm still unconvinced.